app/models/ability.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+class Ability
```

Comment:
Is `app/model` the right place for this? It doesn't look like a model and the CanCanCan documentation just refers to is as a "class" so I was expecting it to be in `lib` but I guess this is where the generator put it?

Reply:
Yes, this is where the generator puts it. I'm happy to put it where ever suits - perhaps in config? I'm not sure about lib, I consider that somewhere where stuff goes that (in theory) could be extracted into a gem, but I can see the logic there since we put various classes into lib already.

From https://github.com/CanCanCommunity/cancancan/blob/develop/lib/generators/cancan/ability/USAGE: 

>The cancan:ability generator creates an Ability class in the models
  directory. You can move this file anywhere you want as long as it
is in the load path.

Reply:
It's tricky... I'm not sure rails even creates a `lib` directory these days, and I know it was removed from the default load but we put it back in `config/application.rb`.

Reply:
Just weighing in, FWIW, Ive seen a folder created for these and seen them put in `app/abilities/` before. It would also make [splitting the ability file](https://github.com/CanCanCommunity/cancancan/wiki/Defining-Abilities:-Best-Practices#split-your-abilityrb-file) (which are likely to want to do later) much easier.

Reply:
`/lib` is almost certainly the wrong place, as this isn't external / library code.  I could see putting it in config, but it's active code that gets checked, and config isn't the first place I'd look.  Ben's thought about app/abilities appeals, if the plan is to deconstruct the large object at some point ...

Reply:
I did do some googling before and there seems to be approximately no consensus on where to put things that aren't one of the "standard" types of class.

I'd always seen `lib` as being for random bits of non-classifiable code and not just for external things (which wouldn't normally be in the repo at all) and in the early days when `lib` was on the autoload path I think that was a more common view but then it got removed.

I didn't even know could just create random directories under `app` to be honest - how does that work? Does rails add every directory there to the autoload path?

Reply:
It does.

there are _some_ additional app directories that are something approaching a standard:  `app/presenters`, `app/forms`, `app/services`  (especially the last one) but it's pretty much open season for organizing things under the top layer of app.  Rails will loop through those directories and add them to autoload at boot time (so, if you add a _new_ `app/foo` class, you'd need to reboot your dev server, but otherwise, classes underneath are subject to Rails' loading magicks)